### PR TITLE
add extruder switch test [CURA-6317]

### PR DESCRIPTION
**Pull request by rburema**

### CODE REVIEW

File: suite_Cura/shared/steps/CommonSteps.py [hunk 1]
```diff
@@ -107,3 +107,27 @@ def step(context):
...
+@Step("I close the extruder selector")
+def step(context):
+    cura.selectExtruderTab("0") # NOTE: Only if already open!
+
+# @Then("I select the [1-8] extruder", regexp=True)
+@When("I select the |any| extruder")
+def step(context, extruder_nr):
+    cura.selectExtruderTab(extruder_nr)
+
+@Step("The |any| nozzle is of type '|any|'")
```
Comment (Ghostkeeper) [inline suite_Cura/shared/steps/CommonSteps.py]: The other steps don't start with capital letters (except for names like `Cura` and the special case of `I`).

File: suite_Cura/tst_Printer_Management/test.feature [hunk 2]
```diff
@@ -1,7 +1,19 @@
...
     Given Cura has been started with preset configurations
+    And I select the 'Ultimaker 3' printer and '0.1' profile
+    When The 1 nozzle is of type 'AA 0.4'
+    And The 2 nozzle is of type 'AA 0.4'
+	And I select the 2 extruder
+	And I select printcore 'AA 0.8'
+	And I show all settings
+	Then The setting 'Layer Height' in 'Quality' is '0.2'
+	And The setting 'Line Width' in 'Quality' is '0.75'
+	And I close the extruder selector
```
Comment (Ghostkeeper) [inline suite_Cura/tst_Printer_Management/test.feature]: Mixing tabs and spaces here.

File: suite_Cura/shared/scripts/PageObjects/PrinterSettingsPage.py [hunk 3]
```diff
@@ -69,6 +68,41 @@ def selectProfileFromPreferences(self, profile_list, profile):
...
+        # Open tab:
+        obj_tab = {"container": names.settings_scrollview, "text": tab_name, "type": "Label", "unnamed": 1, "visible": True}
+        waitForObject(obj_tab)
+        self.click(obj_tab)
+
+        # Find setting-item:
+        obj_label = self.findObjectWithText({"container": names.settings_scrollview, "type": "Label", "unnamed": 1, "visible": True}, setting_name, exact_match=True)
+        child_list = self.getChildrenOfType(object.parent(obj_label), "QQuickTextInput")
+
+        # Close the tab again, so it can be reopened next time (workaround because it's hard to know if it's collapsed or not):
```
Comment (Ghostkeeper) [inline suite_Cura/shared/scripts/PageObjects/PrinterSettingsPage.py]: Yeah, this is a problem for more cases. I think we should eventually remove all of our steps that just click on a button to open a dialogue, and swap them for steps that "ensure that a dialogue is open" just like how we ensure that Cura is running. That would be more semantic in the BDD and more robust for the actual test as well.

Just something to keep in mind for next time. Let's not change all that now.

File: suite_Cura/shared/steps/CommonSteps.py (hunk 1 above)
Comment (Ghostkeeper) [reply suite_Cura/shared/steps/CommonSteps.py]: Fixed here: 766fdb295d66aea1f2c85eae79eaf7251bcfe96f

File: suite_Cura/tst_Printer_Management/test.feature (hunk 2 above)
Comment (Ghostkeeper) [reply suite_Cura/tst_Printer_Management/test.feature]: Fixed here: 5b5779f8c201ef6dfeaf36c8e0420762e4cbafcb

File: suite_Cura/shared/scripts/PageObjects/PrinterSettingsPage.py (hunk 3 above)
Comment (rburema) [reply suite_Cura/shared/scripts/PageObjects/PrinterSettingsPage.py]: > Let's not change all that now.

Should we make a ticket for it then?

Comment (Ghostkeeper) [reply suite_Cura/shared/scripts/PageObjects/PrinterSettingsPage.py]: Yeah let's discuss that with @Ellecross tomorrow.

**Pull request merged**